mergin/client_pull.py

```diff
@@ -229,18 +239,20 @@ def download_project_finalize(job):
             traceback_lines = traceback.format_exception(type(exc), exc, exc.__traceback__)
             job.mp.log.error("Error while downloading project: " + "".join(traceback_lines))
             job.mp.log.info("--- download aborted")
-            job.failure_log_file = _cleanup_failed_download(job.directory, job.mp)
+            job.failure_log_file = _cleanup_failed_download(job.directory.name, job.mp)
             raise future.exception()
 
     job.mp.log.info("--- download finished")
 
     for task in job.update_tasks:
         # right now only copy tasks...
-        task.apply(job.directory, job.mp)
+        task.apply(job.mp.dir, job.mp)
 
     # final update of project metadata
     job.mp.update_metadata(job.project_info)
 
+    job.directory.cleanup()
```

Comment:
is it necessary to call cleanup() explicitly if we create it with `delete=True` ?

Reply:
it should not necessary, but it is seems reasonable to cleanup here. It will happen automatically later, using garbage collector but the moment will be unpredictable. So I would prefer having it there. Also we had it previously for `push_project_finalize()`

Reply:
```suggestion
    job.temp_directory.cleanup()
```

Reply:
okay, let's keep it (btw. normally, objects should be cleaned up immediately by python due to reference counting, only in case of circular references the garbage collection may be delayed)